docs/plugins.md

```diff
@@ -1,3 +1,155 @@
 # Plugins
 
-TBD
+Plugin is a JavaScript package that extends built-in React Native CLI features. It can provide an array of additional commands to run or platforms to target.
+
+For example, `react-native-windows` package is a plugin that provides `react-native run-windows` command and `windows` platform.
+
+Details of this particular integration as well as how to provide an additional platform for React Native were described in a [`dedicated section`](./platforms.md) about platforms.
+
+## How does it work?
+
+Except for React Native dependencies, where configuration is implicit, each package needs to have a `react-native.config.js` at the root folder in order to be discovered by the CLI as a plugin.
+
+```js
+module.exports = {
+  commands: [
+    {
+      name: 'foo-command',
+      func: () => console.log('It worked')
+    }
+  ]
+};
+```
+
+At the startup, React Native CLI reads configuration from all dependencies listed in `package.json` and reduces them into a single configuration. 
+
+At the end, an array of commands concatenated from all plugins is passed on to the CLI to be loaded after built-in commands.
+
+## Command interface
+
+#### `name: string`
+
+Name that will be used in order to run the command. 
+
+Note: If you want your command to accept additional arguments, make sure to include them in the name.
+
+For example, `my-command <argument>` will require an argument to be provided and will throw a validation error otherwise. Alternatively, `my-command [argument]` will accept an argument, but will not throw when run without it. In that case, make sure to check for its presence.
+
+#### `func: (argv: string[], config: ConfigT, options: Object) => ?Promise<void>`
```

Comment:
I think it would be more readable to move the type to a code block and leave the section title to be just `#### func`

Reply:
I don't know. Feels to me that referring to "last argument" or "first argument" in that case would be less understandable than it is right now.

Reply:
either way, I did it.